### ISSUE TITLE
[8.0]Problem on task with latest update from repo

### ISSUE THREAD

**Issue by flotho** [odoo/odoo#8094]
Hi Folks,

I have this error when trying to display the tasks : 
```
AssertionError: Table alias name project_category__parent_id__parent_id__parent_id__parent_id__parent_id is longer than the 64 characters size accepted by default in postgresql.
```
Issue on the latest update either on git .
Commit 02954ba succeed, and 4359f65 failed

**Comment by flotho** [comment]
tested with the rpm https://nightly.odoo.com/8.0/nightly/rpm/odoo_8.0.20150810.noarch.rpm, it works,
with the latest it fails

**Comment by NL66278** [comment]
I have a similar problem. A solution was in the past proposed on launchpad but never implemented. See: https://code.launchpad.net/~initos.com/openobject-server/trunk-bug-1314037/+merge/217575

**Comment by NL66278** [comment]
Problem solved in: https://github.com/odoo/odoo/pull/8142 (launchpad solution + extra problem solved).

**Comment by ccarly** [comment]
I have a similar problem. and patch not solve this

'''
  File "/data/home/openerp/bin/8.0/openerp/models.py", line 4587, in _generate_order_by_inner
    inner_clauses = self._generate_m2o_order_by(alias, order_field, query, do_reverse)
  File "/data/home/openerp/bin/8.0/openerp/models.py", line 4564, in _generate_m2o_order_by
    reverse_direction=reverse_direction)
  File "/data/home/openerp/bin/8.0/openerp/models.py", line 4587, in _generate_order_by_inner
    inner_clauses = self._generate_m2o_order_by(alias, order_field, query, do_reverse)
  File "/data/home/openerp/bin/8.0/openerp/models.py", line 4564, in _generate_m2o_order_by
    reverse_direction=reverse_direction)
  File "/data/home/openerp/bin/8.0/openerp/models.py", line 4587, in _generate_order_by_inner
    inner_clauses = self._generate_m2o_order_by(alias, order_field, query, do_reverse)
  File "/data/home/openerp/bin/8.0/openerp/models.py", line 4564, in _generate_m2o_order_by
    reverse_direction=reverse_direction)
  File "/data/home/openerp/bin/8.0/openerp/models.py", line 4587, in _generate_order_by_inner
    inner_clauses = self._generate_m2o_order_by(alias, order_field, query, do_reverse)
  File "/data/home/openerp/bin/8.0/openerp/models.py", line 4564, in _generate_m2o_order_by
    reverse_direction=reverse_direction)
  File "/data/home/openerp/bin/8.0/openerp/models.py", line 4587, in _generate_order_by_inner
    inner_clauses = self._generate_m2o_order_by(alias, order_field, query, do_reverse)
  File "/data/home/openerp/bin/8.0/openerp/models.py", line 4553, in _generate_m2o_order_by
    dest_model = self.pool[order_field_column._obj]
RuntimeError: maximum recursion depth exceeded while calling a Python object

'''

**Comment by NL66278** [comment]
@ccarly Use pdb or other means to determine which field is causing the problem. My patch guards against a table directly to itself (like category has parent category). In the rare/extrem case a file indirectly refers to itself it would not work. And then you might have a different problem altogether.

**Comment by ccarly** [comment]
Promen solved in #8114 and closed with #7349e23
